### PR TITLE
Refine OIDC sample

### PR DESCRIPTION
**Pull request by troydai**
1. Add signout remote scenario
2. Use bootstrap to enhance the view
3. Improve readability

Effect:
screen shot 2016-08-17 at 3 02 07 pm

### CODE REVIEW

File: samples/OpenIdConnect.AzureAdSample/Startup.cs [hunk 1]
```diff
@@ -103,46 +105,89 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
                 if (context.Request.Path.Equals("/signout"))
                 {
                     await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
-                    context.Response.ContentType = "text/html";
-                    await context.Response.WriteAsync($"<html><body>Signing out {context.User.Identity.Name}<br>{Environment.NewLine}");
-                    await context.Response.WriteAsync("<a href=\"/\">Sign In</a>");
-                    await context.Response.WriteAsync($"</body></html>");
-                    return;
+                    await WriteHtmlAsync(context.Response,
+                        response => response.WriteAsync($"<h1>Signed out locally: {context.User.Identity.Name}</h1><a class=\"btn btn-primary\" href=\"/\">Sign In</a>"));
```
Comment (Tratcher) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: Keep the data dump and the links separated

File: samples/OpenIdConnect.AzureAdSample/Startup.cs [hunk 2]
```diff
@@ -103,46 +105,89 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
...
-                context.Response.ContentType = "text/html";
-                await context.Response.WriteAsync($"<html><body>Hello Authenticated User {context.User.Identity.Name}<br>{Environment.NewLine}");
-                await context.Response.WriteAsync("Claims:<br>" + Environment.NewLine);
-                foreach (var claim in context.User.Claims)
+                else if (context.Request.Path.Equals("/remote-signedout"))
                 {
-                    await context.Response.WriteAsync($"{claim.Type}: {claim.Value}<br>{Environment.NewLine}");
+                    await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
+                    await WriteHtmlAsync(context.Response,
+                        response => response.WriteAsync($"<h1>Signed out remotely: {context.User.Identity.Name}</h1><a class=\"btn btn-primary\" href=\"/\">Sign In</a>"));
```
Comment (Tratcher) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: Split

**Comment by Tratcher** [comment]
Stylish. Planning to do this to any of the other samples? Just doing one would be out of place.
:shipit:

**Comment by troydai** [comment]
Yup. I'll refurbish other samples as well.

File: samples/OpenIdConnect.AzureAdSample/Startup.cs (hunk 1 above)
Comment (troydai) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: will update.

File: samples/OpenIdConnect.AzureAdSample/Startup.cs (hunk 2 above)
Comment (troydai) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: will do.

**Comment by PinpointTownes** [comment]
Looks nice :smile: 

One remark tho: I know these samples are more "sandboxes" than real samples, but I think it's worth adding a tiny comment indicating that nothing is HTML-encoded and that copying the code as-is in a real world app is not recommended.

**Comment by blowdart** [comment]
Or you could actually encode properly ...

**Comment by troydai** [comment]
Add html encoding for user names. I left out the claims' types and values and access token. I assume these are not user input and expect them to be save. Concerns?

File: samples/OpenIdConnect.AzureAdSample/Startup.cs [hunk 3]
```diff
@@ -103,47 +106,93 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
                 if (context.Request.Path.Equals("/signout"))
                 {
                     await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
-                    context.Response.ContentType = "text/html";
-                    await context.Response.WriteAsync($"<html><body>Signing out {context.User.Identity.Name}<br>{Environment.NewLine}");
-                    await context.Response.WriteAsync("<a href=\"/\">Sign In</a>");
-                    await context.Response.WriteAsync($"</body></html>");
-                    return;
+                    await WriteHtmlAsync(context.Response,
+                        response => response.WriteAsync($"<h1>Signed out locally: {HtmlEncode(context.User.Identity.Name)}</h1><a class=\"btn btn-primary\" href=\"/\">Sign In</a>"));
```
Comment (Tratcher) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: Split

File: samples/OpenIdConnect.AzureAdSample/Startup.cs [hunk 4]
```diff
@@ -103,47 +106,93 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
...
+                        await response.WriteAsync("<h2>Tokens:</h2>");
+                        try
+                        {
+                            // Use ADAL to get the right token
+                            var authContext = new AuthenticationContext(authority, AuthPropertiesTokenCache.ForApiCalls(context, CookieAuthenticationDefaults.AuthenticationScheme));
+                            var credential = new ClientCredential(clientId, clientSecret);
+                            string userObjectID = context.User.FindFirst("http://schemas.microsoft.com/identity/claims/objectidentifier").Value;
+                            var result = await authContext.AcquireTokenSilentAsync(resource, credential, new UserIdentifier(userObjectID, UserIdentifierType.UniqueId));
+
+                            await response.WriteAsync($"<h3>access_token</h3><code>{result.AccessToken}</code><br>");
```
Comment (Tratcher) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: No, foreign content must be encoded

File: samples/OpenIdConnect.AzureAdSample/Startup.cs [hunk 5]
```diff
@@ -103,47 +106,93 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
...
+                await response.WriteAsync($"<th>{column}</th>");
+            }
+            await response.WriteAsync("</tr>");
+            foreach (var row in data)
+            {
+                await response.WriteAsync("<tr>");
+                foreach (var column in row)
                 {
-                    await context.Response.WriteAsync($"AquireToken error: {ex.Message}<br>{Environment.NewLine}");
+                    await response.WriteAsync($"<td>{column}</td>");
```
Comment (Tratcher) [inline samples/OpenIdConnect.AzureAdSample/Startup.cs]: Encode these too

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**